File: server/src/main/java/com/family/genealogy/entity/GenerationPoetry.java
package com.family.genealogy.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 字辈诗/字辈歌实体类
 * 用于管理不同姓氏分支的字辈序列及其释义
 */
@Data
@TableName("sys_generation_poetry")
public class GenerationPoetry {
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 姓氏
     */
    private String surname;

    /**
     * 分支名称
     */
    private String branchName;

    /**
     * 字辈诗/字辈歌标题
     */
    private String title;

    /**
     * 字辈序列（按世代顺序排列）
     */
    private String generationSequence;

    /**
     * 字辈释义
     */
    private String interpretation;

    /**
     * 起始世代
     */
    private Integer startGeneration;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

    /**
     * 是否启用
     */
    private Boolean enabled;

    /**
     * 创建人ID
     */
    private Long creatorId;

    /**
     * 备注
     */
    private String remark;
}

File: admin/src/api/generationPoetry.ts
import request from './request'

export interface GenerationPoetry {
  id?: number
  surname: string
  branchName: string
  title: string
  generationSequence: string
  interpretation: string
  startGeneration: number
  enabled: boolean
  creatorId?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  current: number
  size: number
}

// 字辈诗API
export const generationPoetryApi = {
  // 分页查询字辈诗
  getPage: (params: {
    page: number
    pageSize: number
    surname?: string
    branch?: string
  }) => {
    return request.get<PageResult<GenerationPoetry>>('/generation-poetry', { params })
  },

  // 获取所有启用的字辈诗
  getEnabledList: () => {
    return request.get<GenerationPoetry[]>('/generation-poetry/enabled')
  },

  // 根据姓氏获取字辈诗
  getBySurname: (surname: string) => {
    return request.get<GenerationPoetry[]>(`/generation-poetry/by-surname/${surname}`)
  },

  // 获取字辈诗详情
  getById: (id: number) => {
    return request.get<GenerationPoetry>(`/generation-poetry/${id}`)
  },

  // 创建字辈诗
  create: (data: GenerationPoetry) => {
    return request.post<GenerationPoetry>('/generation-poetry', data)
  },

  // 更新字辈诗
  update: (id: number, data: GenerationPoetry) => {
    return request.put<GenerationPoetry>(`/generation-poetry/${id}`, data)
  },

  // 删除字辈诗
  delete: (id: number) => {
    return request.delete(`/generation-poetry/${id}`)
  },

  // 启用/禁用字辈诗
  toggleStatus: (id: number, enabled: boolean) => {
    return request.put(`/generation-poetry/${id}/enable`, {}, {
      params: { enabled }
    })
  }
}